### PR TITLE
Remove isDeployed/isCancelled flags based on lastAction

### PR DESCRIPTION
**Pull request by gskrobisz**
Changes:
1. removed sealed trait `Proces` (that is the most controversial part of this change)
2. removed `Process.isDeployed/isCancelled` methods (we should not evaluate scenario status from lastAction)
3. removed unnecessary method `DeploymentServiceImpl.checkStateInDeploymentManager`
4. side effect: we see what we are looking at (process property, action type flag, last action type flag or status type flag)

StateStatus flags "isXYZ" are removed in https://github.com/TouK/nussknacker/pull/4350

### CODE REVIEW

File: designer/server/src/test/scala/pl/touk/nussknacker/ui/process/deployment/DeploymentServiceSpec.scala [hunk 1]
```diff
@@ -116,7 +116,7 @@ class DeploymentServiceSpec extends AnyFunSuite with Matchers with PatientScalaF
 
     val processDetails = fetchingProcessRepository.fetchLatestProcessDetailsForProcessId[Unit](id).dbioActionValues.value
     processDetails.lastStateAction should not be None
-    processDetails.isCanceled shouldBe true
+    processDetails.lastAction.exists(_.isCanceled) shouldBe true
```
Comment (lciolecki) [inline designer/server/src/test/scala/pl/touk/nussknacker/ui/process/deployment/DeploymentServiceSpec.scala:119]: There should be used: `lastStateAction`

**Review by arkadius — APPROVED** [review]
Why it is still a draft?

File: designer/server/src/test/scala/pl/touk/nussknacker/ui/process/deployment/DeploymentServiceSpec.scala [hunk 2]
```diff
@@ -359,7 +356,7 @@ class DeploymentServiceSpec extends AnyFunSuite with Matchers with PatientScalaF
 
     val processDetails = fetchingProcessRepository.fetchLatestProcessDetailsForProcessId[Unit](id).dbioActionValues.value
     processDetails.lastStateAction shouldBe None
-    processDetails.isNotDeployed shouldBe true
+    processDetails.lastAction shouldBe None
```
Comment (lciolecki) [inline designer/server/src/test/scala/pl/touk/nussknacker/ui/process/deployment/DeploymentServiceSpec.scala:359]: Here and below also should be lastStateAction

**Review by lciolecki — APPROVED** [review]
Good change!

File: designer/server/src/test/scala/pl/touk/nussknacker/ui/process/deployment/DeploymentServiceSpec.scala (hunk 2 above)
Comment (gskrobisz) [reply designer/server/src/test/scala/pl/touk/nussknacker/ui/process/deployment/DeploymentServiceSpec.scala:359]: Here we expect both lastStateAction and lastAction to be `None`.

File: extensions-api/src/main/scala/pl/touk/nussknacker/engine/api/deployment/ProcessAction.scala [hunk 3]
```diff
@@ -12,13 +12,10 @@ import java.util.UUID
                                     processVersionId: VersionId,
                                     performedAt: Instant,
                                     user: String,
-                                    action: ProcessActionType,
+                                    actionType: ProcessActionType,
```
Comment (arkadius) [inline extensions-api/src/main/scala/pl/touk/nussknacker/engine/api/deployment/ProcessAction.scala:15]: Please mention about this change in migration guide - it is a part of the api

Comment (gskrobisz) [reply extensions-api/src/main/scala/pl/touk/nussknacker/engine/api/deployment/ProcessAction.scala:15]: Fixed.

File: designer/server/src/main/scala/pl/touk/nussknacker/ui/process/repository/ProcessDBQueryRepository.scala [hunk 4]
```diff
@@ -112,7 +112,7 @@ object ProcessDBQueryRepository {
     processVersionId = actionData._1.processVersionId.getOrElse(throw new AssertionError(s"Process version not available for finished action: ${actionData._1}")),
     performedAt = actionData._1.performedAtTime.getOrElse(throw new AssertionError(s"PerformedAt not available for finished action: ${actionData._1}")),
     user = actionData._1.user,
-    action = actionData._1.action,
+    actionType = actionData._1.action,
```
Comment (arkadius) [inline designer/server/src/main/scala/pl/touk/nussknacker/ui/process/repository/ProcessDBQueryRepository.scala:115]: We should change action->actionType in DB as well

Comment (gskrobisz) [reply designer/server/src/main/scala/pl/touk/nussknacker/ui/process/repository/ProcessDBQueryRepository.scala]: Fixed.

**Pull request merged**